packages/asset-swapper/src/utils/swap_quote_calculator.ts

```diff
@@ -239,99 +246,155 @@ export class SwapQuoteCalculator {
-                        constants.ZERO_AMOUNT,
-                        remainingTakerAssetFillAmount.minus(takerAssetAmountWithFees),
-                    ),
-                };
-            },
-            {
-                totalMakerAssetAmount: constants.ZERO_AMOUNT,
-                totalTakerAssetAmount: constants.ZERO_AMOUNT,
-                totalFeeTakerAssetAmount: constants.ZERO_AMOUNT,
-                remainingTakerAssetFillAmount: takerAssetSellAmount,
-            },
-        );
+            } else {
+                // This is a collapsed bridge order.
+                // Because collapsed bridge orders actually fill at different rates,
+                // we can iterate over the uncollapsed fills to get the actual
+                // asset amounts transfered.
+                // We can also assume there are no fees and the order is not
+                // partially filled.
+
+                // Infer the bridge slippage from the difference between the fill
+                // size and the atual order asset amounts.
+                const makerAssetBridgeSlippage = !worstCase
+                    ? constants.ONE_AMOUNT
+                    : order.makerAssetAmount.div(order.fill.totalMakerAssetAmount);
+                const takerAssetBridgeSlippage = !worstCase
+                    ? constants.ONE_AMOUNT
+                    : order.takerAssetAmount.div(order.fill.totalTakerAssetAmount);
+                // Consecutively fill the subfills in this order.
+                const subFills = !worstCase ? order.fill.subFills : order.fill.subFills.slice().reverse();
+                for (const subFill of subFills) {
+                    if (remainingTakerAssetFillAmount.minus(takerAssetAmount).lte(0)) {
+                        break;
+                    }
+                    const partialTakerAssetAmount = subFill.takerAssetAmount.times(takerAssetBridgeSlippage);
+                    const partialMakerAssetAmount = subFill.makerAssetAmount.times(makerAssetBridgeSlippage);
+                    const partialTakerAssetFillAmount = BigNumber.min(
+                        partialTakerAssetAmount,
+                        remainingTakerAssetFillAmount.minus(takerAssetAmount),
+                    );
+                    const partialMakerAssetFillAmount = partialTakerAssetFillAmount
+                        .div(partialTakerAssetAmount)
+                        .times(partialMakerAssetAmount)
+                        .integerValue(BigNumber.ROUND_DOWN);
+                    takerAssetAmount = takerAssetAmount.plus(partialTakerAssetFillAmount);
+                    makerAssetAmount = makerAssetAmount.plus(partialMakerAssetFillAmount);
+                }
+            }
+            totalMakerAssetAmount = totalMakerAssetAmount.plus(makerAssetAmount);
+            totalTakerAssetAmount = totalTakerAssetAmount.plus(takerAssetAmount);
+            totalFeeTakerAssetAmount = totalFeeTakerAssetAmount.plus(feeTakerAssetAmount);
```

Comment:
TODO: start to think about when fees are in different assets rather than a constant assumption.

If we cross pairs for a better price fees might be whack in multiple tokens. If orders come from different Relayers they could use a different Fee structure (mF = mA or mF = tA).